### ISSUE TITLE
Unable to assign unique name to Dynamic providers in Pulumi Python

### ISSUE THREAD

**Issue by Ralnoc** [pulumi/pulumi#3468]
While building dynamic providers, it would be extremely helpful to be able to identify which provider is actually implementing an action being taken. Unfortunately, at least in the Pulumi Python, all Dynamic Providers are typed as `pulumi-python:dynamic:Resource`. 

First, I wanted to check to see if I missed something that would allow us to change that identifier. However if it is currently not possible, it would be extremely helpful to be able to do so.

**Comment by pgavlin** [comment]
It is not currently possible to change that identifier, no.

However, it would be a relatively easy change to allow you to specify the last piece of the type (i.e. `Random`). Would that be sufficient for your use case?

**Comment by pgavlin** [comment]
(we can't change the first or second pieces of the type token because they're used by the plugin system to locate which plugin to use for the resource)

**Comment by Ralnoc** [comment]
Absolutely! Allow us to assign a unique name to the final name for the type would work great.

**Comment by schollii** [comment]
@pgavlin I'm not familiar with the pulumi code base but I've programed in python and many other languages for many years so I might be able to help implement this. If you can put a link to a file where this label is currently implemented, I can have a look.

**Comment by Asamsig** [comment]
This was feature implemented [here](https://github.com/pulumi/pulumi/pull/7633).
See this [comment](https://github.com/pulumi/pulumi/pull/7633#issuecomment-920472853) for further confirmation.

So this issue can be closed.

**Comment by iwahbe** [comment]
Fixed by https://github.com/pulumi/pulumi/pull/7633
